vendor/k8s.io/klog/v2/k8s_references.go

```diff
@@ -31,11 +33,30 @@ type ObjectRef struct {
 
 func (ref ObjectRef) String() string {
 	if ref.Namespace != "" {
-		return fmt.Sprintf("%s/%s", ref.Namespace, ref.Name)
+		var builder strings.Builder
+		builder.Grow(len(ref.Namespace) + len(ref.Name) + 1)
+		builder.WriteString(ref.Namespace)
+		builder.WriteRune('/')
+		builder.WriteString(ref.Name)
+		return builder.String()
```

Comment:
I'm surprised this is more efficient than `return ref.Namespace + "/" + ref.Name`

Reply:
The performance difference wasn't big, but the number of allocations went down:

https://github.com/kubernetes/klog/commit/a1638bf6333adeb9e466a8af27ac3e926c31d2e1

But that was a comparison against the [original `fmt.Sprintf`](https://github.com/kubernetes/klog/blame/7ea6d6adb64592611d05066e927aaec7d2d20b16/klog.go#L1628)...

It didn't occur to me to try with simple string concatenation. Let me check.

Reply:
> But that was a comparison against the [original `fmt.Sprintf`](https://github.com/kubernetes/klog/blame/7ea6d6adb64592611d05066e927aaec7d2d20b16/klog.go#L1628)...

yeah, it doesn't surprise me it's cheaper than Sprintf :-)

Reply:
There's a later optimization that avoids the memory allocation entirely when formatting an ObjectReference by calling WriteText instead, so I went back to a commit before that change (https://github.com/kubernetes/klog/commit/a1638bf6333adeb9e466a8af27ac3e926c31d2e1) and tried with string concatenation there.

I'm not sure what magic the Go compiler does to optimize `strings.Builder`, but I get basically the same results for both:
```
name                               old time/op    new time/op    delta
KlogOutput/KObjSlice_okay-36         5.31µs ± 9%    5.35µs ± 8%   ~     (p=0.971 n=10+10)
KlogOutput/KObjSlice_nil_entry-36    4.51µs ± 8%    4.61µs ± 9%   ~     (p=0.289 n=10+10)
KlogOutput/KObj-36                   3.51µs ± 6%    3.51µs ± 9%   ~     (p=1.000 n=10+10)
KlogOutput/KObjSlice_nil_arg-36      3.31µs ± 2%    3.35µs ± 5%   ~     (p=0.315 n=8+10)
KlogOutput/KObjs-36                  4.50µs ± 6%    4.32µs ± 8%   ~     (p=0.063 n=10+10)
KlogOutput/KObjSlice_int_arg-36      4.61µs ± 7%    4.69µs ± 8%   ~     (p=0.393 n=10+10)
KlogOutput/KObjSlice_ints-36         5.76µs ±13%    5.87µs ± 6%   ~     (p=0.565 n=10+10)

name                               old alloc/op   new alloc/op   delta
KlogOutput/KObjSlice_okay-36           714B ± 0%      714B ± 0%   ~     (all equal)
KlogOutput/KObjSlice_nil_entry-36      513B ± 0%      513B ± 0%   ~     (all equal)
KlogOutput/KObj-36                     361B ± 0%      361B ± 0%   ~     (all equal)
KlogOutput/KObjSlice_nil_arg-36        304B ± 0%      304B ± 0%   ~     (all equal)
KlogOutput/KObjs-36                    545B ± 0%      545B ± 0%   ~     (all equal)
KlogOutput/KObjSlice_int_arg-36        561B ± 0%      561B ± 0%   ~     (all equal)
KlogOutput/KObjSlice_ints-36           794B ± 0%      794B ± 0%   ~     (all equal)

name                               old allocs/op  new allocs/op  delta
KlogOutput/KObjSlice_okay-36           16.0 ± 0%      16.0 ± 0%   ~     (all equal)
KlogOutput/KObjSlice_nil_entry-36      12.0 ± 0%      12.0 ± 0%   ~     (all equal)
KlogOutput/KObj-36                     8.00 ± 0%      8.00 ± 0%   ~     (all equal)
KlogOutput/KObjSlice_nil_arg-36        9.00 ± 0%      9.00 ± 0%   ~     (all equal)
KlogOutput/KObjs-36                    12.0 ± 0%      12.0 ± 0%   ~     (all equal)
KlogOutput/KObjSlice_int_arg-36        11.0 ± 0%      11.0 ± 0%   ~     (all equal)
KlogOutput/KObjSlice_ints-36           13.0 ± 0%      13.0 ± 0%   ~     (all equal)
```

Reply:
huh... interesting/surprising